packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -15,23 +15,30 @@ import sentry from '../../lib/sentry';
 
 import * as Amplitude from '../../lib/amplitude';
 import { useCallbackOnce } from '../../lib/hooks';
+import { getMinimumAmount } from 'fxa-shared/subscriptions/stripe';
 
 /*
  * Check if the coupon promotion code provided by the user is valid.
  * If it is valid, return the discounted amount in cents.
  */
 const checkPromotionCode = async (planId: string, promotionCode: string) => {
   try {
-    const { discount } = await apiInvoicePreview({
+    const { discount, total, currency } = await apiInvoicePreview({
       priceId: planId,
       promotionCode,
     });
 
     if (!discount?.amount) {
       throw new Error('No discount for coupon');
     }
+
+    if (total < getMinimumAmount(currency) && total !== 0) {
+      throw new Error('Total below minimum amount allowed.');
```

Comment:
I know that this error is currently not shown the end users.  And I don't think we should.  It would be a strange one to actually show users since this is likely a misconfiguration on the price.

I'll file a jira on preventing that misconfiguration.

Reply:
Interesting. That should be the ESLint `curly` rule, IIRC.
https://github.com/mozilla/eslint-plugin-fxa/blob/fbcfb9db9881d644d147fa9461a4273884160553/lib/configs/default.js#L21

The ESLint `plugin:fxa/default` config should set this to `'curly': [2, 'all'],` (where `2` is old-style for "error").
I don't see why this would not be enabled via the following configs:
| ESLint CONFIG | EXTENDS
|:----|:-----
| https://github.com/mozilla/fxa/blob/main/_dev/.eslintrc | `"extends": ["plugin:fxa/recommended"],`
| https://github.com/mozilla/fxa/blob/main/packages/fxa-shared/.eslintrc | `"extends": ["plugin:fxa/recommended"],`
| https://github.com/mozilla/eslint-plugin-fxa/blob/fbcfb9db9881d644d147fa9461a4273884160553/lib/configs/recommended.js | `extends: ['plugin:fxa/default', 'prettier']`
| https://github.com/mozilla/eslint-plugin-fxa/blob/fbcfb9db9881d644d147fa9461a4273884160553/lib/configs/default.js | `extends: ['eslint:recommended'],`

And yet, looking at ESLint's `--parse-config` for packages/fxa-shared/subscriptions/stripe.ts, it looks like the `curly` rule is disabled:
```sh
npx eslint --config=_dev/.eslintrc.json --print-config packages/fxa-shared/subscriptions/stripe.ts | jq '.rules.curly' | npx prettier --parser=json
# [0, "all"]
```

So I think the only other candidate is probably the rule is being disabled by **eslint-config-prettier**:
https://github.com/prettier/eslint-config-prettier/blob/7ee41f151c7b17d2a6d5cc1211b3910712eadc1b/index.js#L10

```js
    "curly": 0,
```

Reply:
hmm does this need to be on the front end at all?  I believe the invoice preview endpoint can handle it.

Reply:
I wrote a semi-long winded response and then deleted it because I think I should move back a few steps: what's this validation for?  If it's just to show some message on the front end but still allow the subscription (is it still an error then?), then sure let's keep it on the front end.

Reply:
I would say it is an error, but in the context of the Coupon.

It won't block the subscription from going through, but it will block the coupon from being applied.